### ISSUE TITLE
TSQL: Missing External Table and External Provider Declarations

### ISSUE THREAD

**Issue by shaunraj** [sqlfluff/sqlfluff#5548]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following scripts report unparseable sections:

```
DROP USER IF EXISTS [Test User];
CREATE USER [Test User] FROM EXTERNAL PROVIDER;
```

This script links a database principal to a user within AD (https://learn.microsoft.com/en-us/sql/t-sql/statements/create-user-transact-sql?view=sql-server-ver16)

```
IF OBJECT_ID('dbo.TestTable') IS NOT NULL
BEGIN
    DROP EXTERNAL TABLE [dbo].[TestTable];
END
CREATE EXTERNAL TABLE [dbo].[TestTable] ([TestColumn] VARCHAR(30)) 
WITH (data_source = dsTestDataSource schema_name = 'dbo', object_name = 'TestTable');
```

[Source](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-external-table-transact-sql?view=sql-server-ver16&tabs=dedicated)

### Expected Behaviour

The scripts above parse and linting rules apply

### Observed Behaviour

Error occurs when linting. `Unparseable sections detected...`

### How to reproduce

Run `sqlfluff lint` on the above scripts

### Dialect

TSQL

### Version

2.3.5 - sqlfluff
3.9.6 - python

### Configuration

[sqlfluff]
dialect = tsql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by nathanthorell** [comment]
The CREATE EXTERNAL TABLE support exists already in 3.3.0 (it's been in place since https://github.com/sqlfluff/sqlfluff/pull/4642) but your example has invalid keywords of schema_name, object_name in the WITH block.  It should use LOCATION as specified in https://learn.microsoft.com/en-us/sql/t-sql/statements/create-external-table-transact-sql.
```sql
CREATE EXTERNAL TABLE [dbo].[TestTable] ([TestColumn] VARCHAR(30))
WITH (
    DATA_SOURCE = dsTestDataSource,
    LOCATION = 'dbo.TestTable'
);
```
This above corrected example does parse correctly.
